### PR TITLE
#219: wrong executable extension linux

### PR DESCRIPTION
**Pull request by mvomiero**
Fixes: #219

**NOTE: work in progress**

#### Approach:
different executable extension priority lists depending from the os.

#### Changes:
Splits the list EXTENSION_PRIORITY in EXTENSION_PRIORITY_WIN and EXTENSION_PRIORITY_WIN to have different priorities for the choice of the executable extension depending from the OS.

### CODE REVIEW

File: cli/src/main/java/com/devonfw/tools/ide/common/SystemPath.java [hunk 1]
```diff
@@ -28,7 +28,9 @@ public class SystemPath {
 
   private final IdeContext context;
 
-  private static final List<String> EXTENSION_PRIORITY = List.of(".exe", ".cmd", ".bat", ".msi", ".ps1", "");
+  private static final List<String> EXTENSION_PRIORITY_WIN = List.of(".exe", ".cmd", ".bat", ".msi", ".ps1", "");
+
+  private static final List<String> EXTENSION_PRIORITY_UNIX = List.of(".exe", "", ".cmd", ".bat", ".msi", ".ps1");
```
Comment (aBega2000) [inline cli/src/main/java/com/devonfw/tools/ide/common/SystemPath.java:33]: I just wonder if "" should be placed in the first place, and also, are the extensions that are only for WINDOWS also relevant to the list for UNIX? Does that also affect the WIN list, in the same sense?

**Review by aBega2000 — COMMENTED** [review]
Thank you for this PR and for finding this bug. I have left a comment which I think should be addressed and researched a bit as well.

**Comment by hohwille** [comment]
I fixed this in https://github.com/devonfw/IDEasy/pull/238/commits/78151534ae74dedb3637f380ffc834a56ded499d in PR #238

Usually I very much prefer that we have small stories and PRs fixing exactly that story but my time is very limited and the pressure is high to get this test-infrastructure available that blocks the entire team, so I assume, my merge will come first and thereby replace some other PRs.

**Comment by hohwille** [comment]
IMHO this PR can now be closed. @mvomiero thanks for creating the issue and this PR. I could fix this very fast based on your work 👍

**Pull request closed without merging**